### PR TITLE
prevent asterix in min_version

### LINKED ISSUE #914: Prevent use of * in strict_min_version
In https://bugzilla.mozilla.org/show_bug.cgi?id=1297752 its pointed out that usage of * in strict_min_version is confusing lots of people (48.* is not compatible with 48.0) and we should do our best to prevent this from happening until we come up with a better plan.

### PR DESCRIPTION
**Pull request by andymckay**
* fixes #914
* doesn't allow * for min version, but does for max version
* ...also @rpl

### CODE REVIEW

**Comment by tofumatt** [comment]
r+

**Comment by rpl** [comment]
:+1:

File: src/schema/manifest-schema.json [hunk 1]
```diff
@@ -17,12 +17,14 @@
...
-                            "type": "string"
+                            "type": "string",
+                            "pattern": "^[0-9]{1,3}(.[a-z0-9]+)+$"
                         },
                         "strict_max_version": {
                             "default": "*",
                             "description": "Maximum version of Gecko to support. Defaults to '*'. (Requires Gecko 45)",
-                            "type": "string"
+                            "type": "string",
+                            "pattern": "^[0-9]{1,3}(.[a-z0-9*]+)+$"
```
Comment (muffinresearch) [inline src/schema/manifest-schema.json]: Is the `.` correct or should that be escaped? This would allow `9#foo` for example.

Comment (tofumatt) [inline src/schema/manifest-schema.json]: Oh shoot, good catch. It should be escaped and there should be a test for it.

Comment (rpl) [inline src/schema/manifest-schema.json]: yeah, @muffinresearch is definitely right, the `.` is supposed to be escaped (and an another "invalid version number" test case should be added to the test, e.g. something similar to the above invalid example `40#0a2`)

File: src/schema/manifest-schema.json [hunk 2]
```diff
@@ -17,12 +17,14 @@
                         "strict_min_version": {
                             "default": "42a1",
                             "description": "Minimum version of Gecko to support. Defaults to '42a1'. (Requires Gecko 45)",
-                            "type": "string"
+                            "type": "string",
+                            "pattern": "^[0-9]{1,3}(.[a-z0-9]+)+$"
```
Comment (muffinresearch) [inline src/schema/manifest-schema.json]: Same thing here with the `.`

File: src/schema/manifest-schema.json [hunk 3]
```diff
@@ -18,13 +18,13 @@
                             "default": "42a1",
                             "description": "Minimum version of Gecko to support. Defaults to '42a1'. (Requires Gecko 45)",
                             "type": "string",
-                            "pattern": "^[0-9]{1,3}(.[a-z0-9]+)+$"
+                            "pattern": "^[0-9]{1,3}(\\.[a-z0-9]+)+$"
```
Comment (tofumatt) [inline src/schema/manifest-schema.json]: I think it should just be the one `\`, not two.

Comment (andymckay) [inline src/schema/manifest-schema.json]: It's in JSON where \ is a valid escape character, so you have to do \\ in the regex, which is annoying.

Comment (tofumatt) [inline src/schema/manifest-schema.json]: Ah okay. Figured it might've been the case but just wanted to make sure.

**Comment by tofumatt** [comment]
r+

**Pull request merged**